x-pack/legacy/plugins/lens/public/app_plugin/app.tsx

```diff
@@ -150,10 +151,7 @@ export function App({
     }
   }, [docId]);
 
-  // Can save if the frame has told us what it has, and there is either:
-  // a) No saved doc
-  // b) A saved doc that differs from the frame state
-  const isSaveable = state.isDirty && (core.application.capabilities.lens.save as boolean);
+  const isSaveable = lastKnownDoc && core.application.capabilities.lens.save;
```

Comment:
This line in particular will create a conflict with https://github.com/elastic/kibana/pull/48041